src/vs/editor/browser/widget/diffEditorWidget2/utils.ts

```diff
@@ -170,3 +170,19 @@ export function animatedObservable(base: IObservable<number, boolean>, store: Di
 function easeOutExpo(t: number, b: number, c: number, d: number): number {
 	return t === d ? b + c : c * (-Math.pow(2, -10 * t / d) + 1) + b;
 }
+
+export function deepMerge<T extends {}>(source1: T, source2: Partial<T>): T {
+	const result = {} as T;
+	for (const key in source1) {
```

Comment:
Are you trying to return a completely new object? If so, I don't think this is properly cloning source1 if it has any nested properties.

Reply:
I'm not trying to return a completely new object, but I'm not touching the original objects. This is also what makes this method much more efficient as deepClone + mixin.